pkg/distributor/distributor.go

```diff
@@ -299,6 +299,10 @@ func (d *Distributor) Push(ctx context.Context, req *client.WriteRequest) (*clie
 
 	if d.limits.AcceptHASamples(userID) && len(req.Timeseries) > 0 {
 		cluster, replica := findHALabels(d.limits.HAReplicaLabel(userID), d.limits.HAClusterLabel(userID), req.Timeseries[0].Labels)
+		// If cluster has been explicitely set to "", it means we only care about the replica label, so use a placeholder value for KV
+		if cluster == "" {
```

Comment:
What if me move this logic to `findHALabels()`? Something like:

```
func findHALabels(replicaLabel, clusterLabel string, labels []client.LabelAdapter) (string, string) {
   // ...
   
   // <explain why we do it>
   if cluster == "" && clusterLabel == "" {
    cluster = fallbackClusterName
  }

   return cluster, replica
}
```

Does it make sense?

Reply:
I made a small change, it makes the loop cleaner, while keeping the breaking logic, does it looks ok ?